crates/solvers/src/domain/eth/chain.rs

```diff
@@ -0,0 +1,40 @@
+use ethereum_types::U256;
+
+/// A supported Ethereum Chain ID.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ChainId {
+    Mainnet = 1,
+    Goerli = 5,
+    Gnosis = 100,
+}
+
+impl ChainId {
+    /// Creates a new chain ID returning an error if it is unsupported.
```

Comment:
Personally, I think you don't need to restate what the types are already saying. We know that `new()` is used as a constructor and we can see it returns a `Result`. I wouldn't have this comment, I don't think it's very informative - the doc comment that you put on the `ChainId` type is sufficient.

Reply:
0016437aa21683b664984c912239a576cd9e945a